### ISSUE TITLE
Create the README linked sections only if there is a file already

### ISSUE THREAD

**Issue by gabipetrovay** [IonicaBizau/blah#30]
Create the "How to contribute" section only if there is a `CONTRIBUTING.md` file. Same with the license or documentation.

Add the extra option (`-a, --all` or `-f, --force` or `-o, --overwrite`) that generates all the linked files automatically (also overwrites them).

To this last comment, there should be a warning if one wants to overwrite an existing file. This module is a good place to start but will not be able to generate everything for everyone. So the users will partially modify some of the generated files. Using a confirmation to overwrite and a forcing parameter could come handy in this case.

**Comment by IonicaBizau** [comment]
This should be done in the template.

In my [`README.ejs`](https://github.com/IonicaBizau/dotfiles/blob/1fcaa9bd2ee57771477afb6c8b50a01194e6714a/blah-templates/README.ejs#L212-L216) template I have:

```ejs
<% // How to contribute %>
<% if (_.pack.blah.show_contributing !== false) { %>
## How to contribute
Have an idea? Found a bug? See [how to contribute][contributing].
<% } %>
```

Probably that can be improved into something like:

```ejs
<% if (_.pack.blah.show_contributing !== false && isThere("CONTRIBUTING.md")) { %>
## How to contribute
Have an idea? Found a bug? See [how to contribute][contributing].
<% } %>
```

Anyways, I always want a `CONTRIBUTING.md` file.

> Add the extra option (-a, --all or -f, --force or -o, --overwrite) that generates all the linked files automatically (also overwrites them).

Since the content is supposed to be generated and since we anyway have git repositories, I don't think there is a use case for it. :thought_balloon:

**Comment by gabipetrovay** [comment]
> Anyways, I always want a CONTRIBUTING.md file.

Well, not everybody wants it. :busts_in_silhouette: 

The most common use case for changing the generated files is to changes `.gitignore` files but random changes in others files are also possible. So, even if you have git, it will make you loose some time reverting to a previous version because you forgot you changed some things. Also not everyone wants to save a copy of the templates in the repo. This causes duplication. They could come from another repo. For example, one change that I have to do is adding the `.blah` directory to the `.gitignore`

> Probably that can be improved into something like

Then why closing this issue already? :smile:

**Comment by IonicaBizau** [comment]
> Well, not everybody wants it. :busts_in_silhouette:

Then, they shouldn't run `blah -c` :joy: 

> Then why closing this issue already? :smile:

Because it should be fixed in the template, not in the blah core. :stuck_out_tongue: 

I'm thinking that maybe having a `~/.blah.json` file for global configurations would be handy enough where we could have an option for overriding the things:

```json
{
   "override": true
}
```

**Comment by gabipetrovay** [comment]
Maybe running `blah` alone, should generate the whole shebang :boom: and `-r`, `-c`, etc is for low lever geeks. Default, is generating all the files possible. Why not a `blag` configuration in the `package.json`?

**Comment by IonicaBizau** [comment]
> Why not a blag configuration in the package.json?

It could be, but it will override the global config. Basically the priority will be:

 1. CLI options
 2. `blah` in `package.json` or maybe `blah.json`
 3. `~/.blah.json`
 4. Default values

:+1:

**Comment by IonicaBizau** [comment]
> Maybe running blah alone, should generate the whole shebang :boom: and -r, -c, etc is for low lever geeks.

@gabipetrovay This will be available in `3.0.0` (already pushed in the feature branch). By running `blah`, the doc files will be created (CONTRIBUTING, README, LICENSE etc) if they are missing. To override possibly existing files, you will use `-f` (or `--force`).

Also, improved the templates like you mentioned. :grin:

**Comment by gabipetrovay** [comment]
:+1:
